src/test/java/employee/management/assignment3/controller/EmployeeManagementControllerUnitTest.java

```diff
@@ -0,0 +1,134 @@
+package employee.management.assignment3.controller;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import employee.management.assignment3.data.repo.EmployeeRepository;
+import employee.management.assignment3.domain.Employee;
+import employee.management.assignment3.service.EmployeeManagementService;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.web.servlet.MockMvc;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static net.bytebuddy.matcher.ElementMatchers.is;
+import static org.hamcrest.Matchers.hasSize;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@WebMvcTest(EmployeeManagementController.class)
+class EmployeeManagementControllerUnitTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+    @Autowired
+    private ObjectMapper mapper;
+    @MockBean
+    private EmployeeManagementService employeeManagementService;
+    @MockBean
+    private EmployeeRepository employeeRepository;
+    @InjectMocks
+    private EmployeeManagementController employeeManagementController;
+
+    @Test
+    void create_Employee_Return_Ok() throws Exception {
+        Employee employee = new Employee();
+        employee.setId(1);
+        employee.setFirstName("binaya");
+        employee.setLastName("kumar");
+        employee.setEmail("bk@gmail.com");
+        Mockito.when(employeeManagementService.addNewEmployee(Mockito.any(Employee.class))).thenReturn(employee);
+        /*MockHttpServletRequestBuilder builder = MockMvcRequestBuilders.post("/contact/").contentType(MediaType.APPLICATION_JSON_VALUE).accept(MediaType.APPLICATION_JSON)
+                .characterEncoding("UTF-8").content(this.mapper.writeValueAsBytes(contact));*/
+        mockMvc.perform(post("/employee").contentType(MediaType.APPLICATION_JSON_VALUE).content(this.mapper.writeValueAsString(employee))).andExpect(status().isOk()).andExpect(jsonPath("$.firstName").value(employee.getFirstName()));
+
+    }
+
+    @Test
+    void get_AllEmployee_ReturnsAllEmployee_Ok() throws Exception {
+
+        List<Employee> employeeList = new ArrayList<>();
+        Employee employee1 = new Employee();
```

Comment:
You can extract the employee object creation to a separate method and reuse in all the places.